src/error_strings.c

```diff
@@ -54,26 +53,30 @@ gchar *failed_string = NULL;
 #define INIT_ERRMSG(errvar, errtxt) \
 	{ \
 		errvar = g_strdup(errtxt); \
-		if (!no_indata_expected_error) { \
-			printf("%s\n", ALLOCATE_ERROR_STRING); \
+		if (!errvar) { \
+			printf("%s: \"%s\"\n", ALLOCATE_ERROR_STRING, errtxt); \
```

Comment:
I think this was a bug. Failure to allocate a later error string would not be caught if the first one, `no_indata_expected_error` had been allocated successfully.